.github/CONTRIBUTING.md

```diff
@@ -1,151 +1,212 @@
+$ npm run dev
+```
 
-The scope should be descriptive. If your commit contained multiple scopes, consider splitting them up appropriately and
-sending individual pull requests should they not be tightly knit together.
+This will run a local server on `http://localhost:1313` that will automatically update whenever you save a file, though
+without our custom syntax highlighting. If you prefer to view the site as it would appear on production, run the
+following commands instead:
 
-There isn't a set list you have to pick from. Choose whatever seems most appropriate and most descriptive.
+```shellsession
+$ npm run build && npm run preview
+```
 
-#### Summary
+This will build the site, execute the highlighting post-processor, and run a local server on `http://localhost:4173`.
 
-Use the summary to provide a succint description of the change.
+#### Committing Your Changes
 
-* Use the imperative, present tense: "change", not "changed" nor "changes". Think of it as if you're giving the codebase
-  the instruction to *do this* in order to achieve the new desired state.
-* don't capitalise the first letter
-* no dot (.) at the end
-* in total no longer than 50 characters
+Once you are happy with your changes and ensured that the page builds without errors, you can commit your changes. We
+have a small set of rules for commit messages, which help both you and us keep track of changes more easily.
 
-Some people use email to receive and send commits around, having long headers makes listing them not very nice. Any
-decent git-aware editor should tell you when you're exceeding these 50 characters.
+A commit message consists of a header, a body, and a footer:
 
-### <a name="commit-body"></a> Commit Message Body
+```txt
+<header>
+<blank line>
+<body>
+<blank line>
+<footer>
+```
 
-Just as in the summary, use the imperative, present tense: "fix", not "fixed", nor "fixes".
+The header is mandatory and should ideally be limited to about 50 characters. GitHub will automatically break the header
+line once it exceeds 72 characters, so consider that the hard limit. If your change is trivial enough, like a typo or
+fixing a broken link, the header is all you need. It should be written in the imperative mood and be descriptive of what
+precisely was changed.
 
-Explain the motivation for the change in the commit message body. This part should explain *why* you are making that
-change and cleanly summarise what was changed for easier viewing in git logs.
+```diff
+- Update README.md
++ readme: fix broken link to support server
+```
 
-Lines should be broken at 72 characters, unless you are including a terminal log dump, then do not break the lines. Any
-decent git-capable editor should do this automatically for you.
+If you find yourself struggling to concisely summarize your changes in the header, you might want to consider splitting
+them up into separate, more atomic commits, perhaps even separate PRs if these changes can be reasonably separated.
```

Comment:
```suggestion
If you find yourself struggling to concisely summarize your changes in the header, consider splitting them up into multiple commits or perhaps even multiple PRs if appropriate.
```

Reply:
oh the irony that i myself struggled to concisely author this paragraph